### PR TITLE
Deduplicate Qt plugins deployment

### PR DESCRIPTION
**Pull request by dantti**
Sorry I added pre-commit locally to make sure clang-format was run and apparently your sources were not in sync with it

### CODE REVIEW

**Comment by dantti** [comment]
I have update with my .clang-format but still a bit different, I can manually fix it by hand :(, but I'd rather wait for your review and input about code style :)

**Comment by TheAssassin** [comment]
This PR is unfortunately not acceptable because one cannot even tell what changes you implemented. You should never just run an autoformatter you like on an entire code base and the PR it. 

Feel free to fix the PR and provide just the changes you want to propose. But, generally, I advise people to open issues first.

**Comment by dantti** [comment]
as I said I thought this project had .clang-format, it's a pain to work with projects that don't, and pre-commit just makes sure maintainers don't need to keed pointing to style issues which I have seen you pointing at on another PR.

I there was a proper style file I could just rebase and the PR would be fine.

**Comment by dantti** [comment]
@TheAssassin done, I didn't bother to remove the using namespace .. which is not in use anymore, I can remove if you like...

**Comment by dantti** [comment]
Basically, most deployers had the same code that I moved into `BasicPluginsDeployer::deploy()` and read what was changing in each deployer with the list from `qtPluginsToBeDeployed`, for the corner cases where there was more logic `BasicPluginsDeployer::customDeploy()` is called with no need to call the base class as the code should go to `BasicPluginsDeployer::deploy()`.

File: src/deployers/BasicPluginsDeployer.cpp
Comment (bjorn) [inline src/deployers/BasicPluginsDeployer.cpp:32]: It's a nice cleanup, but instead of introducing this code here along with the `qtPluginsToBeDeployed` function, I would suggest to introduce a protected `BasicPluginsDeployer::deployPlugins(const std::vector<std::string> &plugins)`, so in each `deploy()` implementation one could just do for example:
```c++
    deployPlugins({"bearer"});
```
I think that's adding a bit less complexity overall, since it removes the need for both `qtPluginsToBeDeployed` and `customDeploy`.

Comment (dantti) [reply src/deployers/BasicPluginsDeployer.cpp:32]: while a `deployStandardQtPlugins()` could be nice indeed having a `customDeploy()` or if going with this function then maybe `doDeploy()` is a better naming, this is because we override methods and current code was calling the base method which makes the code slower due extra branching and is error prone as one has to remember to call that.
So I'll add a new commit and we see what is best.

Comment (bjorn) [reply src/deployers/BasicPluginsDeployer.cpp]: Yeah, that's better and I do think it's nice how `doDeploy` allows to get rid of all the calls to the superclass. Though I would personally just have removed them, along with the base implementation, because "we might add more functionality later on" is not a good reason for all that cruft in my opinion.

Anyway, this patch is a good clean-up as-is in my opinion.

**Review by TheAssassin — APPROVED** [review]
Not a fan of having both `deploy` and `doDeploy` (and I think the "deployStandard" name is a bit misleading), but these things tend to happen during refactorings. It's not really a problem. Looks good to me. I can work with these changes!

In the next iteration, we may just use templating to generate all the necessary deployers on the fly and further reduce the amount of code.

Please note that I just ran a quick smoke test. We need to be prepared for incoming issues.

Thank you so much!

**Comment by TheAssassin** [comment]
Build failure is due to a change in the upstream runtime repo. Fixing this on master, rebasing afterwards.

**Comment by TheAssassin** [comment]
Should be fixed. Let's wait for the results. Thanks to @dantti and all the reviewers. I really do enjoy the outcome.

**Pull request merged**
